diff -r 29247e44df47 stubdom/Makefile
--- a/stubdom/Makefile	Fri Nov 30 21:51:17 2012 +0000
+++ b/stubdom/Makefile	Tue Dec 04 20:47:37 2012 -0500
@@ -112,7 +112,7 @@
 	( cd newlib-$(XEN_TARGET_ARCH) && \
 	  CC_FOR_TARGET="$(CC) $(TARGET_CPPFLAGS) $(TARGET_CFLAGS) $(NEWLIB_CFLAGS)" AR_FOR_TARGET=$(AR) LD_FOR_TARGET=$(LD) RANLIB_FOR_TARGET=$(RANLIB) ../newlib-$(NEWLIB_VERSION)/configure --prefix=$(CROSS_PREFIX) --verbose --target=$(GNU_TARGET_ARCH)-xen-elf --enable-newlib-io-long-long --disable-multilib && \
 	  $(MAKE) DESTDIR= && \
-	  $(MAKE) DESTDIR= install )
+	  $(MAKE) DESTDIR=$(DESTDIR) install )
 
 ############
 # Cross-zlib
@@ -132,7 +132,7 @@
 	( cd $< && \
 	  CFLAGS="$(TARGET_CPPFLAGS) $(TARGET_CFLAGS)" CC=$(CC) ./configure --prefix=$(CROSS_PREFIX)/$(GNU_TARGET_ARCH)-xen-elf && \
 	  $(MAKE) DESTDIR= libz.a && \
-	  $(MAKE) DESTDIR= install )
+	  $(MAKE) DESTDIR=$(DESTDIR) install )
 
 ##############
 # Cross-libpci
