From f8006706f4b1a16a03e6720d4b635b260baea3c3 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Tue, 3 Apr 2012 13:12:34 -0400
Subject: [PATCH 4/4] Fix Makefile - replace LIBEXECDIR

---
 Makefile |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 14c9817..e5eccb2 100644
--- a/Makefile
+++ b/Makefile
@@ -66,7 +66,8 @@ create-files:
 	for i in \
 		48tlp-rdw-lock 49bay 49wwan tlp tlp-nop tlp-rdw-nm \
 		tlp-rdw-udev tlp-rf tlp-stat tlp.upstart tlp.init; do \
-		sed 's/@LIBDIR@/$(LIBDIR)/g' <$$i.in >$$i; \
+		sed -e 's/@LIBDIR@/$(LIBDIR)/g' \
+                    -e 's/@LIBEXECDIR@/$(LIBEXECDIR)/g' <$$i.in >$$i; \
 	done
 
 install: create-files install-tlp install-rdw
-- 
1.7.7.6

