diff -Nru bumblebee-3.0.1.orig/src/bbsecondary.c bumblebee-3.0.1/src/bbsecondary.c
--- bumblebee-3.0.1.orig/src/bbsecondary.c	2012-12-11 15:08:53.126991000 -0500
+++ bumblebee-3.0.1/src/bbsecondary.c	2012-12-11 15:09:33.724484000 -0500
@@ -146,6 +146,7 @@
       XORG_BINARY,
       bb_config.x_display,
       "-config", x_conf_file,
+      "-configdir", "/var/empty",
       "-sharevts",
       "-nolisten", "tcp",
       "-noreset",
