diff -Nru cdu-0.37.orig/Makefile cdu-0.37/Makefile
--- cdu-0.37.orig/Makefile	2012-05-05 01:12:13.552628646 -0400
+++ cdu-0.37/Makefile	2012-05-05 01:16:42.667565556 -0400
@@ -32,15 +32,15 @@
 
 install-strip:
 install: all installdirs
-	$(INSTALL) $(PROG).1 ${DESTDIR}/man/man1
+	$(INSTALL) -m 644 $(PROG).1 ${DESTDIR}/share/man/man1
 	$(INSTALL) -m 755 $(TARGET) ${DESTDIR}/bin
 
 installdirs:
-	$(INSTALL) -d ${DESTDIR}/man/man1
+	$(INSTALL) -d ${DESTDIR}/share/man/man1
 	$(INSTALL) -d ${DESTDIR}/bin
 
 uninstall:
-	rm -f ${DESTDIR}/man/man1
+	rm -f ${DESTDIR}/share/man/man1
 	rm -f ${DESTDIR}/bin
 
 
