From 0410485c17995d03d0ed0ee86a96d255700ab463 Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Mon, 12 Mar 2012 15:32:52 -0400
Subject: [PATCH] Fix theme path for Fedora packaging

---
 src/theme.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/theme.py b/src/theme.py
index 1024b1b..be2f586 100755
--- a/src/theme.py
+++ b/src/theme.py
@@ -57,7 +57,7 @@ class Theme:
     
     def getThemeDir(self):
         '''Get theme directory.'''
-        return "../theme/%s/" % (self.themeName)
+        return "theme/%s/" % (self.themeName)
                 
     def getThemePath(self, path):
         '''Get pixbuf path.'''
-- 
1.7.7.6

