diff -Nru laptop-mode-tools_1.61.orig/etc/laptop-mode/laptop-mode.conf laptop-mode-tools_1.61/etc/laptop-mode/laptop-mode.conf
--- laptop-mode-tools_1.61.orig/etc/laptop-mode/laptop-mode.conf	2012-09-06 16:15:18.147036142 -0400
+++ laptop-mode-tools_1.61/etc/laptop-mode/laptop-mode.conf	2012-09-06 16:30:22.200546628 -0400
@@ -39,15 +39,6 @@
 
 
 ###############################################################################
-# Enable/Disable laptop-mode-tools execution
-# ------------------------------------------
-# Set it to 0 to completely disable laptop-mode-tools from running
-###############################################################################
-#
-ENABLE_LAPTOP_MODE_TOOLS=1
-
-
-###############################################################################
 # Configuration debugging
 # -----------------------
 ###############################################################################
diff -Nru laptop-mode-tools_1.61.orig/man/laptop-mode.conf.8 laptop-mode-tools_1.61/man/laptop-mode.conf.8
--- laptop-mode-tools_1.61.orig/man/laptop-mode.conf.8	2012-09-06 16:15:18.150036140 -0400
+++ laptop-mode-tools_1.61/man/laptop-mode.conf.8	2012-09-06 16:30:47.553533802 -0400
@@ -51,10 +51,6 @@
 disabling, plus the core features: the Linux kernel laptop mode feature and
 related settings.
 
-.IP "\fBENABLE_LAPTOP_MODE_TOOLS\fP" 10
-This module determines if laptop-mode-tools should be run or not. Default is 1
-Set it to 0 if you would like to completely disable laptop-mode-tools
-
 .IP "\fBVERBOSE_OUTPUT\fP" 10
 Set this to 1 if you want to see a lot of output when you start/stop laptop mode, and to 0 if you
 don't want this. Useful for debugging purposes.
diff -Nru laptop-mode-tools_1.61.orig/usr/sbin/laptop_mode laptop-mode-tools_1.61/usr/sbin/laptop_mode
--- laptop-mode-tools_1.61.orig/usr/sbin/laptop_mode	2012-09-06 16:15:18.395035998 -0400
+++ laptop-mode-tools_1.61/usr/sbin/laptop_mode	2012-09-06 16:32:22.432486141 -0400
@@ -128,7 +128,6 @@
 NOLM_AC_BRIGHTNESS_COMMAND=false
 LOG_TO_SYSLOG=1
 DEBUG=0
-ENABLE_LAPTOP_MODE_TOOLS=1
 
 # Initialize the PATH Variable
 export PATH="${PATH}":/bin:/sbin:/usr/bin:/usr/sbin
@@ -247,11 +246,6 @@
 	    exit 1
     fi
 
-    if [ x$ENABLE_LAPTOP_MODE_TOOLS = x0 ]; then
-	    log "MSG" "laptop-mode-tools is disabled in config file. Exiting"
-	    exit 0;
-    fi
-
     # Add a simple bash debug mode switch
     if [ "$DEBUG" -eq 1 ]; then
 	set -vx;
@@ -605,9 +599,6 @@
 	    case "$1" in 
 		    init) INIT=1 ;;
 		    force) FORCE=1 ;;
-		    # Old options. We always do "auto" for any option now, but
-		    # we still have to accept the options.
-		    start) ;;
 		    stop) INITSCRIPT_STOP=1 ;;
 		    auto) ;;
 		    modules=*)
@@ -684,14 +675,6 @@
 	    STATE=disabled
     fi
 
-    if [ "$ACTIVATE" -eq 1 -a -f /etc/default/laptop-mode ] ; then
-	    . /etc/default/laptop-mode
-	    if ! ( echo "$ENABLE_LAPTOP_MODE" |grep y ) ; then
-		    log "VERBOSE" "Not starting laptop mode because it is disabled in /etc/default/laptop-mode."
-		    STATE=disabled
-	    fi
-    fi
-
     if [ "$STATE" = "disabled" ] ; then
 	    ACTIVATE=0
     fi
