diff -Nru VirtualGL-2.3.orig/server/vgllogin VirtualGL-2.3/server/vgllogin
--- VirtualGL-2.3.orig/server/vgllogin	2012-02-25 00:12:07.913553458 -0500
+++ VirtualGL-2.3/server/vgllogin	2012-02-25 00:24:13.739333971 -0500
@@ -37,12 +37,8 @@
 
 	if [ "$2" = "" -o "$3" = "" ]; then usage $0; fi
 	XAUTH=xauth
-	if [ -x /usr/X11R6/bin/xauth ]; then
-		XAUTH=/usr/X11R6/bin/xauth
-	else
-		if [ -x /usr/openwin/bin/xauth ]; then
-			XAUTH=/usr/openwin/bin/xauth
-		fi
+	if [ -x /usr/bin/xauth ]; then
+		XAUTH=/usr/bin/xauth
 	fi
 	if [ -z "$DISPLAY" ]; then
 		DNUM=`echo $2 | sed 's/.*[:]//g'`
